sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -95,5 +95,57 @@ public static bool IsMsTestOrNUnitTestIgnored(this IMethodSymbol method)
                     a.AttributeClass.Is(KnownType.Xunit_FactAttribute) ||
                     a.AttributeClass.Is(KnownType.Xunit_TheoryAttribute) ||
                     a.AttributeClass.Is(KnownType.LegacyXunit_TheoryAttribute));
+
+        public static KnownType FindTestAttribute(this IMethodSymbol method)
```

Comment:
I suggest to rename to `FindFirstTestAttribute` or maybe even to return the collection of test attributes and to filter at the caller place.